src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3519,7 +3519,11 @@ private void VisitTupleExpression(BoundTupleExpression node)
                     this.State[slot] = NullableAnnotation.NotNullable;
                     TrackNullableStateOfTupleElements(slot, tupleOpt, arguments, elementTypes, useRestField: false);
                 }
-                _resultType = TypeSymbolWithAnnotations.Create(tupleOpt.WithElementTypes(elementTypes), NullableAnnotation.NotNullable);
+
+                tupleOpt = tupleOpt.WithElementTypes(elementTypes);
+                var locations = tupleOpt.TupleElements.SelectAsArray(x => x.Locations.IsEmpty ? node.Syntax.Location : x.Locations[0]);
```

Comment:
>SelectAsArray(x => x.Locations.IsEmpty ? node.Syntax.Location : x.Locations[0]) [](start = 55, length = 79)

Use overload that takes additional arg to avoid the closure. And consider using `FirstOrDefault`:
```
SelectAsArray(
    (element, location) => element.Locations.FirstOrDefault() ?? location,
    node.Syntax.Location);
```